src/lib/utils/thorchain/memo.ts

```diff
@@ -2,43 +2,60 @@ import { bn } from 'lib/bignumber/bignumber'
 
 import { THORCHAIN_AFFILIATE_NAME } from './constants'
 
-const assertMemoHasPool = (pool: string, memo: string) => {
+const assertMemoHasPool = (pool: string | undefined, memo: string): pool is string => {
   if (!pool) throw new Error(`pool is required in memo: ${memo}`)
+  return true
 }
```

Comment:
We could use the type assertion syntax here, which I think is the more idiomatic way of doing this.
Then we could remove the `return true` as it'll do it implicitly so long as the type assertion function doesn't throw.

E.g.

```suggestion
const assertMemoHasPool = (pool: string | undefined, memo: string): asserts pool is string => {
  if (!pool) throw new Error(`pool is required in memo: ${memo}`)
}
```

Reply:
8582fbd60c34e06b86104008523d3f7a69a26e54 - note the switch to good ol' function syntax: https://stackoverflow.com/a/72689922